src/test/java/io/cdap/plugin/successfactors/source/SuccessFactorsSourceTest.java

```diff
@@ -0,0 +1,467 @@
+
+    new Expectations(SuccessFactorsUrlContainer.class, SuccessFactorsTransporter.class,
+                     SuccessFactorsSchemaGenerator.class) {
+      {
+        successFactorsUrlContainer.getTesterURL();
+        result = getUrl();
+        minTimes = 0;
+
+        successFactorsUrlContainer.getMetadataURL();
+        result = getUrl();
+        minTimes = 0;
+
+        successFactorsTransporter.callSuccessFactorsEntity(null, anyString, anyString);
+        result = getResponseContainer();
+        minTimes = 0;
+
+        successFactorsSchemaGenerator.buildDefaultOutputSchema(anyString);
+        result = getPluginSchema();
+        minTimes = 0;
+      }
+    };
+
+    successFactorsSource = new SuccessFactorsSource(pluginConfig);
+    Map<String, Object> plugins = new HashMap<>();
+    MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(null, plugins);
+    successFactorsSource.configurePipeline(pipelineConfigurer);
+    schema = pipelineConfigurer.getOutputSchema();
+    Assert.assertNotNull("Output Schema generated is not Null", schema);
+  }
+
+  @Test
+  public void testConfigurePipelineSchemaNull() {
+    SuccessFactorsPluginConfig pluginConfig = pluginConfigBuilder.build();
+    new Expectations(SuccessFactorsPluginConfig.class) {
+      {
+        pluginConfig.isSchemaBuildRequired();
+        result = false;
+        minTimes = 0;
+      }
+    };
+    Map<String, Object> plugins = new HashMap<>();
+    MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(null, plugins);
+    successFactorsSource = new SuccessFactorsSource(pluginConfig);
+    successFactorsSource.configurePipeline(mockPipelineConfigurer);
+    Assert.assertNull(schema);
+  }
+
+  @Test(expected = ValidationException.class)
+  public void testConnectionException() {
+    pluginConfig = pluginConfigBuilder.build();
+    successFactorsSource = new SuccessFactorsSource(pluginConfig);
+    Map<String, Object> plugins = new HashMap<>();
+    MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(null, plugins);
+    successFactorsSource.configurePipeline(mockPipelineConfigurer);
+  }
+
+  @Test
+  public void testConfigurePipelineConnectionException() {
+    pluginConfig = pluginConfigBuilder.build();
+    new Expectations(SuccessFactorsPluginConfig.class, SuccessFactorsTransporter.class) {
+      {
+        pluginConfig.isSchemaBuildRequired();
+        result = true;
+        minTimes = 0;
+      }
+    };
+    try {
+      successFactorsSource = new SuccessFactorsSource(pluginConfig);
+      Map<String, Object> plugins = new HashMap<>();
+      MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(null, plugins);
+      successFactorsSource.configurePipeline(mockPipelineConfigurer);
+      Assert.fail("Connection exception is not thrown if valid connection is provided");
+    } catch (ValidationException te) {
+      List<ValidationFailure> failures = te.getFailures();
```

Comment:
why is validation failure expected here? can you add a comment?

Reply:
Added a comment.